crates/compute/src/cpu/layer.rs

```diff
@@ -119,6 +119,33 @@ impl<F: TowerTop> CpuLayerExecutor<F> {
 			})
 			.collect()
 	}
+
+	fn process_kernels_chunks<R>(
+		&self,
+		map: impl Sync
+		+ for<'a> Fn(
+			&'a mut CpuKernelBuilder,
+			usize,
+			Vec<KernelBuffer<'a, F, CpuMemory>>,
+		) -> Result<R, Error>,
+		mut mem_maps: Vec<KernelMemMap<'_, F, CpuMemory>>,
+	) -> Result<Vec<R>, Error> {
```

Comment:
It's a pity you that your're allocating this vector and actually don't use it in both cases.
Maybe better return `impl Iterator<Item = Result<R, Error>>` ?